api/src/test/java/org/pmiops/workbench/api/WorkspacesControllerTest.java

```diff
@@ -2371,7 +2371,49 @@ public void testShareWorkspace() {
   }
 
   @Test
-  public void testShareWorkspaceAddBillingProjectUser() {
+  public void testShareWorkspacePatch() {
+    stubFcGetGroup();
+    DbUser writerUser = createAndSaveUser("writerfriend@gmail.com", 124L);
+    DbUser readerUser = createAndSaveUser("readerfriend@gmail.com", 125L);
+
+    stubFcGetWorkspaceACL();
+    Workspace workspace = createWorkspace();
+    workspace = workspacesController.createWorkspace(workspace).getBody();
+    ShareWorkspaceRequest shareWorkspaceRequest = new ShareWorkspaceRequest();
+    shareWorkspaceRequest.setWorkspaceEtag(workspace.getEtag());
+
+    addUserRoleToShareWorkspaceRequest(
+        shareWorkspaceRequest, "readerfriend@gmail.com", WorkspaceAccessLevel.READER);
+    addUserRoleToShareWorkspaceRequest(
+        shareWorkspaceRequest, "writerfriend@gmail.com", WorkspaceAccessLevel.WRITER);
+
+    // Simulate time between API calls to trigger last-modified/@Version changes.
+    fakeClock.increment(1000);
+    stubFcUpdateWorkspaceACL();
+    WorkspaceUserRolesResponse shareResp =
+        workspacesController
+            .shareWorkspacePatch(
+                workspace.getNamespace(), workspace.getName(), shareWorkspaceRequest)
+            .getBody();
+
+    verify(mockWorkspaceAuditor).fireCollaborateAction(anyLong(), anyMap());
+
+    Workspace workspace2 =
+        workspacesController
+            .getWorkspace(workspace.getNamespace(), workspace.getName())
+            .getBody()
+            .getWorkspace();
+    assertThat(shareResp.getWorkspaceEtag()).isEqualTo(workspace2.getEtag());
```

Comment:
This is comparing the `workspace` retrieved by "get" to the initially created workspace.  They should be the same.  This also shows that the workspace row didn't update, because sharing is fundamentally a Terra operation.

Maybe this should be `retrievedWorkspace` rather than `workspace2` ?